### PR TITLE
Customize Amplitude logging to minimize extra output in tests and builds

### PR DESCRIPTION
**Pull request by TurnerRiley**
As per [this Slack thread](https://codedotorg.slack.com/archives/C0T0PNTM3/p1682967256364519), it was brought up that our Amplitude events are logging to console even in unit tests (meaning every drone build is printing out a lot of unnecessary information). In addition to this, some developers were finding it cluttering to their browser's console when they weren't specifically working on Amplitude events.

This PR updates our logging behavior so that Amplitude events are only logged to the console if the current environment is `development`.

## Browser console logging
Amplitude now does not print to the console log by default:
FINAL_with_false

Amplitude only logs in development once flag is flipped to 'true' locally:
FINAL_with_true

## Testing
Previously, there were lots of Amplitude log events cluttering up the unit tests:
Sample_testing_amplitude

Now, they aren't logged in the test environment:
No_test_amplitude

## Links
Slack thread this was discussed in: [here](https://codedotorg.slack.com/archives/C0T0PNTM3/p1682967256364519)
Jira task: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?modal=detail&selectedIssue=ACQ-568&assignee=60d62161f65054006980bd71)

## Testing story
Local testing and ensuring the drone build is not printing Amplitude logs when running unit tests.

## PR Checklist:
- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/util/AnalyticsReporter.js [hunk 1]
```diff
@@ -10,10 +10,13 @@ import {
   getEnvironment,
   isProductionEnvironment,
   isStagingEnvironment,
+  isDevelopmentEnvironment,
 } from '../../utils';
 
 // A flag that can be toggled to send events regardless of environment
 const ALWAYS_SEND = false;
+// A flag that can be toggled to console log events in development
+const LOG_EVENT_IN_DEV = false;
```
Comment (bethanyaconnor) [inline apps/src/lib/util/AnalyticsReporter.js]: Curious why @rshipp thinks here but I would still prefer these to be automatically logged in development only. If I'm the only one, though, I think this is okay.

Also tagging @hannahbergam as she's been working with Amplitude a lot in the last few months.

Comment (rshipp) [reply apps/src/lib/util/AnalyticsReporter.js]: I agree, I would prefer it to not log in test envs but log by default in dev envs.

Comment (hannahbergam) [reply apps/src/lib/util/AnalyticsReporter.js]: Agreed! Love having those local logs but would be happy to see them removed from Test.

**Review by bethanyaconnor — APPROVED** [review]
🎉 

One small consideration here (that doesn't block this PR) is that Amplitude logging won't be visible on adhocs anymore. I'm okay going forward with this change then adjusting it if it's causing inconveniences for folks!

**Pull request merged**